nixos/modules/hardware/decklink.nix

```diff
@@ -0,0 +1,24 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.decklink;
+  kernelPackages = config.boot.kernelPackages;
+in
+{
+  options.hardware.decklink.enable = mkEnableOption "Enable hardware support for the Blackmagic Design Decklink audio/video interfaces.";
+
+  config = mkIf cfg.enable {
+    boot.kernelModules = [ "blackmagic" "blackmagic-io" "snd_blackmagic-io" ];
+    boot.extraModulePackages = [ kernelPackages.decklink ];
+
+    # supporting service
+    systemd.services."DecklinkVideoHelper" = {
+      enable = true;
+      after = [ "syslog.target" "local-fs.target" ];
+      wantedBy = [ "multi-user.target" ];
+      script = "${pkgs.blackmagicDesktopVideo}/bin/DesktopVideoHelper -n";
```

Comment:
Can you put this in serviceConfig.ExecStart instead so it doesn't spawn a shell?